src/Enum.php

```diff
@@ -146,6 +146,7 @@ public static function values()
      *
      * @psalm-pure
      * @psalm-suppress ImpureStaticProperty
+     * @psalm-suppress ImpureMethodCall
```

Comment:
Fix following issues:

```

ERROR: ImpureMethodCall - src/Enum.php:158:38 - Cannot call an impure constructor from a pure context (see https://psalm.dev/203)
            $reflection            = new \ReflectionClass($class);


ERROR: ImpureMethodCall - src/Enum.php:159:51 - Cannot call a non-mutation-free method ReflectionClass::getConstants from a pure context (see https://psalm.dev/203)
            static::$cache[$class] = $reflection->getConstants();
```

And adding the `/** @psalm-suppress ImpureMethodCall */` comment annotation to fix above issues.

Here are references are as follows:

- https://psalm.dev/docs/running_psalm/issues/ImpureMethodCall/
- https://github.com/vimeo/psalm/issues/2109